src/Serialization.js

```diff
@@ -161,15 +171,19 @@ exports.newMint = () => lib.Mint.new();
 exports._bigIntToInt = maybe => bigInt => {
   // this is needed because try/catch overuse breaks runtime badly
   // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
-  const str = bigInt.to_str();
-  if (/^-[0-9]+$/.test(str)) {
-  return maybe.just(lib.Int.new_negative(lib.BigNum.from_str(str.slice(1)))
-  );
-  } else if (/^[0-9]+$/.test(str)) {
-    return maybe.just(lib.Int.new(lib.BigNum.from_str(str)));
+  try {
+    check_limit(bigInt, -18446744073709551616, 18446744073709551616); // 2 ^ 64
```

Comment:
Also I am not sure about the exact limits, is it strictly less for positive and less-or-equal for negative? Please add the corner cases to the tests.